File: examples/00_graphics/graphic_colors.py
import math
from codac import *

def show_color(color_func, i):
  box = IntervalVector([math.floor(i / 3), i % 3]).inflate(0.4)
  DefaultFigure.draw_box(box, [color_func(1.0), color_func(1.0)])

i = 0

colors = [
  Color.light_gray, Color.gray, Color.dark_gray,
  Color.light_green, Color.green, Color.dark_green,
  Color.light_blue, Color.blue, Color.dark_blue,
  Color.light_cyan, Color.cyan, Color.dark_cyan,
  Color.light_yellow, Color.yellow, Color.dark_yellow,
  Color.light_orange, Color.orange, Color.dark_orange,
  Color.light_red, Color.red, Color.dark_red,
  Color.light_brown, Color.brown, Color.dark_brown,
  Color.light_purple, Color.purple, Color.dark_purple,
  Color.light_pink, Color.pink, Color.dark_pink,
]

for color_func in colors:
  show_color(color_func, i)
  i += 1

DefaultFigure.set_axes(
  axis(0, [-0.5, (i / 3) - 1 + 0.5]),
  axis(1, [-0.5, 2 + 0.5])
)

DefaultFigure.selected_fig().auto_scale()

File: examples/00_graphics/graphic_colors.cpp
#include <codac>

using namespace std;
using namespace codac2;

void show_color(Color (*color)(float), int i)
{
  DefaultFigure::draw_box(
    IntervalVector({ std::floor(i/3.), (double)(i%3) }).inflate(.4),
    { color(1.), color(1.) });
}

int main()
{
  int i = 0;

  show_color(&Color::light_gray, i++);
  show_color(&Color::gray, i++);
  show_color(&Color::dark_gray, i++);

  show_color(&Color::light_green, i++);
  show_color(&Color::green, i++);
  show_color(&Color::dark_green, i++);

  show_color(&Color::light_blue, i++);
  show_color(&Color::blue, i++);
  show_color(&Color::dark_blue, i++);

  show_color(&Color::light_cyan, i++);
  show_color(&Color::cyan, i++);
  show_color(&Color::dark_cyan, i++);

  show_color(&Color::light_yellow, i++);
  show_color(&Color::yellow, i++);
  show_color(&Color::dark_yellow, i++);

  show_color(&Color::light_orange, i++);
  show_color(&Color::orange, i++);
  show_color(&Color::dark_orange, i++);

  show_color(&Color::light_red, i++);
  show_color(&Color::red, i++);
  show_color(&Color::dark_red, i++);

  show_color(&Color::light_brown, i++);
  show_color(&Color::brown, i++);
  show_color(&Color::dark_brown, i++);

  show_color(&Color::light_purple, i++);
  show_color(&Color::purple, i++);
  show_color(&Color::dark_purple, i++);

  show_color(&Color::light_pink, i++);
  show_color(&Color::pink, i++);
  show_color(&Color::dark_pink, i++);

  DefaultFigure::set_axes(axis(0,{-.5,(i/3)-1+.5}), axis(1,{-.5,2+.5}));
  DefaultFigure::selected_fig()->auto_scale();
}
